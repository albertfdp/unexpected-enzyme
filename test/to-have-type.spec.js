import expect from './unexpected-enzyme';
import { mount } from 'enzyme';
import React from 'react';

const Foo = () => <div>Component</div>;
const Bar = () => <span>Another</span>;

describe('to-have-type', () => {
  describe('string', () => {
    it('passes when actual matches expected', () => {
      expect(mount(<div />), 'to have type', 'div');
    });

    it('passes negated when actual does not match the expected', () => {
      expect(mount(<div />), 'not to have type', 'span');
    });

    it('fails when actual does not match expected', () => {
      expect(
        () => expect(mount(<div />), 'to have type', 'span'),
        'with error matching snapshot'
      );
    });

    it('fails when negated actual matches expected', () => {
      expect(
        () => expect(mount(<div />), 'not to have type', 'div'),
        'with error matching snapshot'
      );
    });
  });

  describe('function', () => {
    it('passes when actual matches expected', () => {
      expect(mount(<Foo />), 'to have type', Foo);
    });

    it('passes negated when actual does not match the expected', () => {
      expect(mount(<Foo />), 'not to have type', Bar);
    });

    it('fails when actual does not match expected', () => {
      expect(
        () => expect(mount(<Foo />), 'to have type', Bar),
        'with error matching snapshot'
      );
    });

    it('fails when negated actual matches expected', () => {
      expect(
        () => expect(mount(<Foo />), 'not to have type', Foo),
        'with error matching snapshot'
      );
    });
  });
});
