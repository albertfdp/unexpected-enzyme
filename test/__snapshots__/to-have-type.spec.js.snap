// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`to-have-type function fails when actual does not match expected 1`] = `
"expected
<Foo>
  <div>
    Component
  </div>
</Foo>
to have type Bar"
`;

exports[`to-have-type function fails when negated actual matches expected 1`] = `
"expected
<Foo>
  <div>
    Component
  </div>
</Foo>
not to have type Foo"
`;

exports[`to-have-type string fails when actual does not match expected 1`] = `
"expected <div /> to have type 'span'

-div
+span"
`;

exports[`to-have-type string fails when negated actual matches expected 1`] = `"expected <div /> not to have type 'div'"`;
